.github/workflows/python-check.yaml

```diff
@@ -36,10 +36,9 @@ jobs:
         uses: PyO3/maturin-action@v1
         with:
           working-directory: icechunk-python
-          args: --release --out dist --find-interpreter
+          args: --release --out dist -i python${{ env.PYTHON_VERSION }}
           sccache: true
-        env:
-          CARGO_TERM_QUIET: true
+          manylinux: auto
```

Comment:
this matches the way we build in the release workflow (using manylinux) but constrains to only build for the PYTHON_VERSION version of python to speed things up